plugins/services/src/js/columns/ServicesTableNameColumn.tsx

```diff
@@ -14,33 +14,70 @@ import Pod from "../structs/Pod";
 import { SortDirection } from "plugins/services/src/js/types/SortDirection";
 import ServiceTableUtil from "../utils/ServiceTableUtil";
 
+const ServiceName = React.memo(
+  ({
+    isFiltered,
+    id,
+    isGroup,
+    name,
+    image,
+    webUrl
+  }: {
+    isFiltered: boolean;
+    id: string;
+    isGroup: boolean;
+    name: string;
+    image: string | null;
+    webUrl: string | null;
+  }) => {
+    const serviceLink = isGroup
+      ? `/services/overview/${id}`
+      : `/services/detail/${id}`;
+
+    return (
+      <TextCell>
+        <div className="service-table-heading flex-box flex-box-align-vertical-center table-cell-flex-box text-overflow">
+          <Link className="table-cell-icon" to={serviceLink}>
+            {getImage(image, isGroup)}
+          </Link>
+          <span className="table-cell-value table-cell-flex-box">
+            {getServiceLink(id, name, isGroup, isFiltered)}
+            {getOpenInNewWindowLink(webUrl)}
+          </span>
+        </div>
+      </TextCell>
+    );
+  }
+);
+
 export function nameRenderer(
   isFiltered: boolean,
   service: Service | Pod | ServiceTree
 ): React.ReactNode {
-  const id: string = encodeURIComponent(service.getId().toString());
-  const isGroup = service instanceof ServiceTree;
-  const serviceLink = isGroup
-    ? `/services/overview/${id}`
-    : `/services/detail/${id}`;
+  // These do not work with instanceof ServiceTree due to TS
+  const image =
+    service instanceof Pod || service instanceof Service
+      ? service.getImages()["icon-small"]
+      : null;
+  const webUrl =
+    service instanceof Pod || (service instanceof Service && hasWebUI(service))
+      ? service.getWebURL()
+      : null;
 
   return (
-    <TextCell>
-      <div className="service-table-heading flex-box flex-box-align-vertical-center table-cell-flex-box text-overflow">
-        <Link className="table-cell-icon" to={serviceLink}>
-          {getImage(service)}
-        </Link>
-        <span className="table-cell-value table-cell-flex-box">
-          {getServiceLink(service, isFiltered)}
-          {getOpenInNewWindowLink(service)}
-        </span>
-      </div>
-    </TextCell>
+    <ServiceName
```

Comment:
'nit: I think it would have been possible to split this into `ServiceName` and `GroupName` component 🤔 not sure if I'm missing something and I think its not that important tho.

Reply:
Same as before, I did not touch anything in this direction. I don't see that this is part of the scope of this ticket.

Reply:
At least if our goal is still to merge this functionality today. If this goal changed we can also close this PR and use the #3792 one